src/db.rs

```diff
@@ -461,15 +461,15 @@ impl Database {
                         let table_pages_iter = AllPageNumbersBtreeIter::new(
                             table_root,
                             definition.get_fixed_key_size(),
-                            <&DynamicCollection>::fixed_width(),
+                            <&DynamicCollection<()>>::fixed_width(),
```

Comment:
Can you introduce a method: `fn fixed_width_with(value_width: Option<usize>) -> Option<usize>` that always returns `None`? and then change this to `DynamicCollection::fixed_width_with(definition.get_fixed_value_size())`. The current code works, but is a bit confusing because the type of the table isn't `DynamicCollection<()>`

Reply:
introduced that method and replaced where relevant 👍